src/pages/staticPages/tools/oncoprinter/OncoprinterImportUtils.ts

```diff
@@ -140,25 +142,47 @@ function getOncoprinterGeneticInputLine(parsed: OncoprinterGeneticInputLine) {
                 line.push('FUSION');
                 break;
         }
+        if (parsed.trackLabel) {
+            line.push(sanitizeColumnData(parsed.trackLabel));
+        }
     }
     return line.join('  ');
 }
 
 export function getOncoprinterGeneticInput(
-    oncoprintData: Pick<GeneticTrackDatum, 'data' | 'sample' | 'patient'>[],
+    oncoprintData: {
+        label: string;
+        data: Pick<GeneticTrackDatum, 'data' | 'sample' | 'patient'>[];
+    }[],
     caseIds: string[],
     sampleOrPatient: 'sample' | 'patient'
 ) {
-    const parsedLines = [];
-    for (const oncoprintDatum of oncoprintData) {
-        parsedLines.push(
-            ...oncoprintDatum.data.map(d => {
-                return getOncoprinterParsedGeneticInputLine(d, oncoprintDatum[
-                    sampleOrPatient
-                ] as string);
+    // handle case where tracks have same label
```

Comment:
This looks like a good candidate to isolate into a utility function (`generateUniqNames`?) and add a few tests.

Reply:
Agree!